versioned/tiered/tiered-impl/src/main/java/com/dremio/nessie/versioned/impl/InternalRef.java

```diff
@@ -98,16 +98,17 @@ InternalTag getTag() {
    * Implement {@link Ref} to build an {@link InternalRef} object.
    */
   // Needs to be a package private class, otherwise class-initialization of ValueType fails with j.l.IllegalAccessError
-  static final class Builder extends EntityBuilder<InternalRef> implements Ref {
+  static final class Builder extends EntityBuilder<InternalRef> implements Ref, Ref.Tag, Ref.Branch {
```

Comment:
I think it would be cleaner for all of these implements of Ref to actually be nested structures as opposed to implementing all the interfaces.

Reply:
Implementations are actually free to either return `this` or a new instance.
E.g. while all implementations of `Ref.tag()` or `Ref.branch()` currently return `this`, others may choose to return some specialized `TagBuilder` or `BranchBuilder`.